File: e2e/playwright-python/providers/support_provider.py
from playwright.sync_api import Page
from support.api.api_client import ApiClient
from support.mocks.wiremock_client import WireMockClient
from support.mocks.api_route_mock import ApiRouteMock


class MockProvider:
    def __init__(self, page: Page):
        self.api = ApiRouteMock(page)


class SupportProvider:
    def __init__(self, page: Page):
        self.api = ApiClient()
        self.wiremock = WireMockClient()
        self.mock = MockProvider(page)

File: e2e/cucumber-playwright-typescript/providers/SupportProvider.ts
import { APIRequestContext, Page } from '@playwright/test';
import { ApiClient } from '@/support/api/ApiClient';
import { WireMockClient } from '@/support/mocks/WireMockClient';
import { ApiRouteMock } from '@/support/mocks/ApiRouteMock';

export class SupportProvider {
  readonly api: ApiClient;
  readonly wiremock: WireMockClient;
  readonly mock: { api: ApiRouteMock };

  constructor(page: Page, request: APIRequestContext) {
    this.api = new ApiClient(request);
    this.wiremock = new WireMockClient(request);
    this.mock = { api: new ApiRouteMock(page) };
  }
}
